LocalNotifications/Sources/LocalNotifications/LocalNotifications.swift

```diff
@@ -0,0 +1,136 @@
+                self.permitted = true
+                onComplete.callDeferred()
+                return
+            }
+
+            do {
+                GD.print("Requesting Authorisation")
+                let authorizationGranted = try await notificationCenter.requestAuthorization(options: authorizationOptions)
+                self.permitted = authorizationGranted
+                onComplete.callDeferred()
+            } catch {
+                GD.print("Authorisation request failed. \(error.localizedDescription)")
+                self.permitted = false
+            }
+        }
+    }
+    
+    @Callable
+    func isPermitted() -> Bool {
+        self.permitted
+    }
+
+    @Callable
+    func wasNotificationAccepted() -> Bool {
+        self.notificationAccepted
+    }
+
+    @Callable
+    func removeAllNotifications() {
+        let notificationCenter = UNUserNotificationCenter.current()
+        let pending = notificationCenter.removeAllPendingNotificationRequests()
+    }
+
+    @Callable
+    func removeNotification(uid:String) {
+        let notificationCenter = UNUserNotificationCenter.current()
+        notificationCenter.removePendingNotificationRequests(withIdentifiers: [uid])
+    }
+
+    /**
+        Request that a notification banner be displayed in the future, as long
+        as the user is not inside the application. `reqeustPermission()` must
+        be called first.
+
+        - parameter uid: Random string or identifier used to delete this notification later if needed
+        - parameter title: title of the local notification.
+        - parameter body: text body of the local notification.
+        - parameter seconds: display notification in how many seconds time.
+        - returns no value is instantly expected. onComplete read the variable `notificationAccepted`
+     */
+    @Callable
+    func schedule(_ uid:String, _ title:String, _ body:String, _ seconds:Int, _ onComplete:Callable) {
+        Task {
+            if await hasPendingNotification(uid:uid) {
+                GD.print("Notification already exists with UID \(uid). Rescheduling.")
+                await removeNotification(uid:uid)
+            }
+            self.notificationAccepted = false
+
+            let content = UNMutableNotificationContent()
+            content.title = title
+            content.body = body
+            content.sound = UNNotificationSound.default
+        
+            let trigger = UNTimeIntervalNotificationTrigger(timeInterval:  TimeInterval(seconds), repeats: false)
+            let request = UNNotificationRequest(identifier: uid, content: content, trigger: trigger)
+
+            do {
+                try await UNUserNotificationCenter.current().add(request)
+                GD.print("Scheduled notification \(uid).")
+                notificationAccepted = true
+                onComplete.callDeferred()
+            } catch {
+                GD.print("Error scheduling: \(error.localizedDescription)")
```

Comment:
I would probably use GD.pushError() to handle errors, unless the error is sometimes expected, in which case I would use GD.pushWarning() instead. Even though it doesn't matter as much if you can't test it in the editor

Reply:
I was using GD.print() to cause information to display in the Xcode console, to allow debugging/testing inside Xcode.  I'll remove most of the print statements, except for true errors that we would would want to see in the Xcode console.

Reply:
GD.pushError also show up in the Xcode console with a big "USER ERROR" in front, I'm not sure about GD.pushWarning but I assume it's similar so I would keep those around for when needed